icu4c/source/common/umutablecptrie.cpp

```diff
@@ -18,6 +18,11 @@
 #include "uassert.h"
 #include "ucptrie_impl.h"
 
+// ICU-20235 In case Microsoft math.h has defined this, undefine it.
+#ifdef OVERFLOW
```

Comment:
I don't remember -- does undef complain if the macro was not defined?

Reply:
Not sure off-hand about other compilers, but with MSVC it doesn't complain:
> You can also apply the #undef directive to an identifier that has no previous definition. This ensures that the identifier is undefined.

Reply:
I don't know of any compiler that actually requires the #ifdef/#undef incantation, but it's traditional as a way to express that a symbol might in some cases or in some environments be defined, and it's used in a number of places throughout the ICU4C code base.